pkg/registries/registries_test.go

```diff
@@ -878,14 +875,12 @@ func TestEditRegistriesConfig(t *testing.T) {
 		})
 	}
 
-	// should fail if idms or itms exist with icsp
-	failtests := []struct {
+	successTests := []struct {
```

Comment:
I was really making an argument about the structure of the tests. Previously, there was exactly one `Test…` unit test function per test subject; and the `TestEditRegistriesConfig` was two table-driven test loops, one for success, one for failure.

The changes in this PR are I don’t know why we would have two separate table-driven loops in `TestEditRegistriesConfig` instead of just entries in a single table-driven test loop, and I don’t know why some specific cases, which are not _that_ unusual, are in their own little `TestEdit…` functions. A table-driven test with a single table entry… is not so table-driven.

Now, I fully appreciate that the test data in `TestEdit…` is huge and unmanageable; it just seems to me that making another copy of the test code, and less sharing instead of more sharing, does not help with the manageability at all.

Reply:
Ahhh. Thank you for the clarification. I merged the redundant functions into TestEditRegistriesConfig. The function is not really manageable, but will leave it to a further PR to refactor that test case.